### PR TITLE
Replace Data.BigInt with JS.BigInt

### LINKED ISSUE #1472: Switch from `Data.BigInt` to `JS.BigInt`
- https://github.com/purescript-contrib/purescript-bigints that uses `big-integer` NPM package is deprecated
- https://github.com/purescript-contrib/purescript-js-bigints uses native bigints

### PR DESCRIPTION
**Pull request by kozross**
Closes #1472. Due to us needing Aeson instances, I had to create a newtype wrapper: I've tried to keep its API as similar to the `Data.BigInt` one as possible, so that switching is painless. 

For some reason, I get CI failures due to Nix not installing the necessary library: locally I have no problems.

### Pre-review checklist

- [X] All code has been formatted using our config (`make format`)
- [X] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [X] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: src/Internal/Types/BigInt.purs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import Data.Maybe (Maybe(Nothing, Just))
+import Data.Maybe (fromMaybe)
+import Data.Reflectable (class Reflectable)
+import Data.String.NonEmpty.Internal (NonEmptyString(NonEmptyString))
+import JS.BigInt as BigIntImpl
+import Prim (Boolean, Int, Number, String, Symbol)
+import Prim.Int (class ToString)
+import Type.Proxy (Proxy)
+
+newtype BigInt = UnsafeBigInt BigIntImpl.BigInt
```
Comment (klntsky) [inline src/Internal/Types/BigInt.purs:41]: Is there any other reason other than Aeson to have a newtype? If not, please just make `aeson` package depend on `js-bigints` (and "undepend" it from `bigints`) and put the instances near the aeson typeclasses. Then remove this newtype wrapper. The reason is: we want to make it easier to integrate with the rest of the purescript ecosystem libraries - they know about js-bigints, but working with our BigInt would require (un)wrapping and will lead to more confusion.

**Review by klntsky — CHANGES_REQUESTED** [review]

**Comment by klntsky** [comment]
> For some reason, I get CI failures due to Nix not installing the necessary library: locally I have no problems.

You probably forgot to run spago2nix, read `doc/development.md`

File: src/Internal/Types/BigInt.purs (hunk 1 above)
Comment (kozross) [reply src/Internal/Types/BigInt.purs:41]: That was the only reason really. I'll send a PR to `aeson`.

File: src/Contract/Numeric/BigInt.purs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import Prelude
+
+import JS.BigInt (BigInt)
+import JS.BigInt as BigInt
+
+abs :: BigInt -> BigInt
+abs x = case compare x (BigInt.fromInt 0) of
+  LT -> negate x
+  _ -> x
```
Comment (klntsky) [inline src/Contract/Numeric/BigInt.purs:12]: This is in Prelude. Please remove this module entirely.

https://pursuit.purescript.org/packages/purescript-prelude/6.0.1/docs/Data.Ord#v:abs

File: packages.dhall [hunk 3]
```diff
@@ -60,6 +60,7 @@ Syntax:
 where `<version>` is:
 - a tag (i.e. "v4.0.0")
 - a branch (i.e. "master")
+import Data.Maybe (fromMaybe)
```
Comment (klntsky) [inline packages.dhall:63]: Accidental paste?

**Review by klntsky — CHANGES_REQUESTED** [review]

**Comment by bladyjoker** [comment]
@klntsky @kozross I fixed the CI error as I would like to upgrade LB to this whole new shiny thing asap

**Comment by bladyjoker** [comment]
> @klntsky @kozross I fixed the CI error as I would like to upgrade LB to this whole new shiny thing asap

```
  ✗ AdditionalUtxos example:
  Error: Couldn't get tx output for (TransactionInput { index: 1u, transactionId: (TransactionHash (hexToByteArrayUnsafe "a95adbfc463c87aca6eeab66174bf803752d0b861643dcd26e2bbd6c451f43ac")) })
    at Module.error (file:///build/output/Effect.Exception/foreign.js:6:10)
    at <anonymous> (/nix/store/yh2q4qc6dvfm6vxjf90gk4i7jv4vxm82-cardano-transaction-lib-src/src/Internal/Contract/MinFee.purs:79:7)
    at go (file:///build/output/Data.Traversable/foreign.js:36:42)
    at file:///build/output/Data.Traversable/foreign.js:46:20
    at <anonymous> (/build/.spago/foldable-traversable/v6.0.0/src/Data/Traversable.purs:207:20)
    at <anonymous> (/nix/store/yh2q4qc6dvfm6vxjf90gk4i7jv4vxm82-cardano-transaction-lib-src/src/Internal/Contract/MinFee.purs:118:59)
    at <anonymous> (/nix/store/yh2q4qc6dvfm6vxjf90gk4i7jv4vxm82-cardano-transaction-lib-src/src/Internal/Contract/MinFee.purs:78:44)
    ```

**Comment by bladyjoker** [comment]
> > @klntsky @kozross I fixed the CI error as I would like to upgrade LB to this whole new shiny thing asap
> 
> ```
>   ✗ AdditionalUtxos example:
>   Error: Couldn't get tx output for (TransactionInput { index: 1u, transactionId: (TransactionHash (hexToByteArrayUnsafe "a95adbfc463c87aca6eeab66174bf803752d0b861643dcd26e2bbd6c451f43ac")) })
>     at Module.error (file:///build/output/Effect.Exception/foreign.js:6:10)
>     at <anonymous> (/nix/store/yh2q4qc6dvfm6vxjf90gk4i7jv4vxm82-cardano-transaction-lib-src/src/Internal/Contract/MinFee.purs:79:7)
>     at go (file:///build/output/Data.Traversable/foreign.js:36:42)
>     at file:///build/output/Data.Traversable/foreign.js:46:20
>     at <anonymous> (/build/.spago/foldable-traversable/v6.0.0/src/Data/Traversable.purs:207:20)
>     at <anonymous> (/nix/store/yh2q4qc6dvfm6vxjf90gk4i7jv4vxm82-cardano-transaction-lib-src/src/Internal/Contract/MinFee.purs:118:59)
>     at <anonymous> (/nix/store/yh2q4qc6dvfm6vxjf90gk4i7jv4vxm82-cardano-transaction-lib-src/src/Internal/Contract/MinFee.purs:78:44)
>     ```
> ```

Dammit this passes locally! Oh God of Reproducibility, where is you universal determinism now? Why does Plutip have free will while the rest of us are wallowing in our destinies?

**Comment by bladyjoker** [comment]
Woohoo! @klntsky this is green!

**Review by klntsky — APPROVED** [review]

**Pull request merged**